### DISCUSSION TITLE
Allow a parser option to customize storage format

### DISCUSSION THREAD

**Discussion by mathiashsteffensen** [mashpie/i18n-node#489] [Ideas]
Hello,

First of all, thank you for creating this awesome library. I would love to use it but far prefer the YAML based storage that I am used to from web frameworks like Ruby on Rails.

I see in the README it is on the roadmap to implement a storage adapter to allow for customizing this. Is this still the plan?

If so I think i have suggestion. It isn't quite a storage adapter, but looking at the code, if you added a 'parser' option which defaults to
```javascript
var parser = opt.parser || JSON
```

It would allow customization of the parser to be any object that responds to .parse and .stringify, this would make it really easy to plug in for example a YAML parser https://eemeli.org/yaml/v1/#parse-amp-stringify

What do you think of this idea? I would love to contribute and whip up a PR so you would only have to review it :)

**Comment by mashpie** [comment]
This indeed seems to provide a nice quick win as it doesn't require extra dependencies. If you'd like to submit a PR I'll be happy to review it.

**Comment by mathiashsteffensen** [comment]
Thank you for allowing me to contribute :pray:, I have submitted a PR now
